src/fsharp/TypedTreeOps.fs

```diff
@@ -3200,7 +3222,12 @@ let isByrefLikeTyconRef (g: TcGlobals) m (tcref: TyconRef) =
     | _ -> 
        let res = 
            isByrefTyconRef g tcref ||
-           (isStructTyconRef tcref && TyconRefHasAttribute g m g.attrib_IsByRefLikeAttribute tcref)
+           (match isStructTyconRef tcref, TyconRefHasAttributeByName m tname_IsByRefLikeAttribute tcref with
+            | true, hasAttr -> hasAttr
+            | false, hasAttr ->
+                if hasAttr then
+                    errorR(Error(FSComp.SR.tcByRefLikeNotStruct(), tcref.Deref.Range))
```

Comment:
Usually we do not raise errors in functions like this - the check should go back in PostInferenceChecks.

Reply:
Done @TIHan.